build.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+arch=($1)
+
+case ${arch} in
+    i386 ) base_image="resin/i386-alpine" ;;
+    amd64 ) base_image="alpine:3.7" ;;
+    arm ) base_image="resin/rpi-alpine" ;;
+    aarch64 ) base_image="resin/aarch64-alpine" ;;
+esac
+cp Dockerfile Dockerfile.${arch}
+sed -i "s@alpine:\([0-9]\+\).\([0-9]\+\)@$base_image@g" Dockerfile.${arch}
```

Comment:
I run shellcheck on it, here is a "fixed" version:

```bash
#!/bin/bash

set -euo pipefail

arch="$1"

case "$arch" in
    i386 ) base_image="resin/i386-alpine" ;;
    amd64 ) base_image="alpine:3.7" ;;
    arm ) base_image="resin/rpi-alpine" ;;
    aarch64 ) base_image="resin/aarch64-alpine" ;;
esac

sed "s@alpine:\\([0-9]\\+\\).\\([0-9]\\+\\)@$base_image@g" Dockerfile > "Dockerfile.$arch"
```

Are you ok with it?

- The first line is defensive (stop on all errors, stop on undeclared vars, stop on failure in pipe), I _always_ use this line in bash scripts.
- vars have been quoted
- `\` have been `\\`
- cp+sed merged into one (no edit-in-place)

Reply:
Ah nice, I'll pinch that for my own repo's. Looks much more concise and all makes sense.

Reply:
Will you include the modification?
If not, I can commit it then merge your PR.

Reply:
Yeah just doing it now (was on my phone before)